src/main/java/org/testng/internal/Invoker.java

```diff
@@ -474,109 +387,101 @@ else if (! m_continueOnFailedConfiguration) {
 
     // check if there are failed @BeforeGroups
     String[] groups = method.getGroups();
-    if(null != groups && groups.length > 0) {
-      for(String group : groups) {
-        if(m_beforegroupsFailures.containsKey(group)) {
-          result = false;
-          break;
-        }
+    for (String group : groups) {
+      if (m_beforegroupsFailures.containsKey(group)) {
+        result = false;
+        break;
       }
     }
     return result;
   }
 
-   // Creates a token for tracking a unique invocation of a method on an instance.
-   // Is used when configFailurePolicy=continue.
-  private static Object getMethodInvocationToken(ITestNGMethod method, Object instance) {
-    return String.format("%s+%d+%d", instance.toString(), method.getCurrentInvocationCount(), method.getParameterInvocationCount());
+  private boolean isBeforeClassMethod(ITestNGMethod currentTestMethod) {
+    return currentTestMethod != null && m_methodInvocationResults.containsKey(currentTestMethod);
```

Comment:
I suppose it was the previous behavior but the relation between the method name and the business code is not easy.

Reply:
Fixed.